### ISSUE TITLE
[Suggestion]: Dimensional/Portal Crafting

### ISSUE THREAD

**Issue by Lanse505** [xt9/InWorldCrafting#9]
![image](https://user-images.githubusercontent.com/1495068/53304948-87ae7100-387b-11e9-9feb-ff342d13d9af.png)

TLDR;
The ability to "craft" one item or possibly a set of items (might be harder).
To another item by throwing it into a portal.

This could be handled by grabbing the EntityTeleportEvent or DimensionChanged or smth.
(don't remember the name right now).

**Comment by Lanse505** [comment]
So I was thinking about the player event which is wrong.
However as I figured there is a generic entity event.

EntityTravelToDimensionEvent:
![image](https://user-images.githubusercontent.com/1495068/53305115-af9ed400-387d-11e9-9de2-8f22bbee785b.png)



This should give you an Entity object to query against and a dimension it's traveling to.
Something you could do is if you want to allow things like "Alfenheim" portal interactions.
Is to Cull the Entity Itemstack going in, Spawn a converted Entity ItemStack that comes out and cancel the event.

Though that's just for fancies :P

**Comment by Lanse505** [comment]
Also as always if you want help/a PR just give me a poke and I'll try to set one up :P

**Comment by xt9** [comment]
Yeah this could be interesting. I'm not sure if it would be viable for multiple inputs because timing would be a PITA and you would have to check surrounding of the item, but as a single input type of craft I think this is cool.

**Comment by Lanse505** [comment]
Agreed
